recipes/libspng/all/patches/0.7.4-0001-allow-miniz.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index da1917e..30c220d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -12,6 +12,7 @@ set(SPNG_VERSION ${SPNG_MAJOR}.${SPNG_MINOR}.${SPNG_REVISION})
+ option(ENABLE_OPT "Enable architecture-specific optimizations" ON)
+ option(SPNG_SHARED "Build shared lib" ON)
+ option(SPNG_STATIC "Build static lib" ON)
++option(SPNG_USE_MINIZ "Use Miniz instead of zlib" OFF)
+ option(BUILD_EXAMPLES "Build examples" ON)
+ 
+ include(GNUInstallDirs)
+@@ -47,14 +48,24 @@ if(SPNG_STATIC)
+     list(APPEND spng_TARGETS spng_static)
+ endif()
+ 
+-find_package(ZLIB REQUIRED)
++if(SPNG_USE_MINIZ)
++    find_package(miniz REQUIRED)
++    set(compress_LIBS miniz::miniz)
++else()
++    find_package(ZLIB REQUIRED)
++    set(compress_LIBS ZLIB::ZLIB)
++endif()
```

Comment:
I might be missing some historical reasons in this recipe or library, but why if the library uses just zlib, why injecting this option to use miniz? I'd say drop the ``with_miniz`` option and go with the library ``zlib`` implemented one.

Reply:
@RubenRBS @jcar87 
Sorry for the late response.
The reason is as @jcar87 wrote about miniz support. (Thanks @jcar87)
I will try to create a PATCH with libspng.

If that is not acceptable, close this PR and create another PR to build with meson.
This is because I want to provide the recipe with as many official options as possible.

Reply:
Hi @toge ! THanks for your reply - I believe the PR is fine since it overcomes a buildsystem limitation, we just didn't have enough context. Perhaps you could elaborate a little more in the `patch_description` to say that miniz is supported, but is currently missing support in the cmake scripts - and link either my comment above: https://github.com/conan-io/conan-center-index/pull/17618#discussion_r1242166287, or even better, an issue raised upstream (doesn't need to be a full fledged patch, but we greatly encourage raising issues first in the upstream libraries rather than fixing them here directly).

This traceability can help future maintainers remove the patches once they are no longer needed, and also encourages improvements all around :D Thanks for your contribution!